polyfill/lib/calendar.mjs

```diff
@@ -333,53 +332,1354 @@ function resolveNonLunisolarMonth(calendarDate) {
+    },
+    adjustCalendarDate(calendarDate /*, cache, overflow, fromLegacyDate = false */) {
+      // Because this is not a lunisolar calendar, it's safe to convert monthCode to a number
+      const { month, monthCode } = calendarDate;
+      if (month === undefined) calendarDate = { ...calendarDate, month: monthCodeNumberPart(monthCode) };
+      this.validateCalendarDate(calendarDate);
+      calendarDate = this.completeEraYear(calendarDate);
+      calendarDate = nonIsoHelperBase.adjustCalendarDate(calendarDate);
+      return calendarDate;
+    },
+    estimateIsoDate(calendarDate) {
+      calendarDate = this.adjustCalendarDate(calendarDate);
+      const { year, month, day } = calendarDate;
+      const { anchorEra } = this;
+      const isoYearEstimate = year + anchorEra.isoEpoch.year - (anchorEra.hasYearZero ? 0 : 1);
+      return ES.RegulateDate(isoYearEstimate, month, day, 'constrain');
+    },
+    // Several calendars based on the Gregorian calendar use Julian dates (not
+    // proleptic Gregorian dates) before the Julian switchover in Oct 1582. See
+    // https://bugs.chromium.org/p/chromium/issues/detail?id=1173158.
+    checkJulianBug(calendarDate, isoDate) {
+      if (this.vulnerableToJulianBug) {
+        const beforeJulianSwitch = this.compareCalendarDates(isoDate, { year: 1582, month: 10, day: 15 }) < 0;
+        if (beforeJulianSwitch) {
+          const testDate = new Date('+001001-01-01T00:00Z').toLocaleDateString(`en-US-u-ca-${this.id}`, {
+            timeZone: 'UTC'
+          });
+          if (testDate.startsWith('12')) {
+            throw new RangeError(
+              `calendar '${this.id}' is broken for ISO dates before 1582-10-15` +
+                ' (see https://bugs.chromium.org/p/chromium/issues/detail?id=1173158)'
+            );
+          }
+        }
+      }
+    },
+    vulnerableToJulianBug: false
+  });
+};
+
+const makeHelperOrthodox = (id, originalEras) => {
+  const base = makeHelperGregorian(id, originalEras);
+  return ObjectAssign(base, {
+    inLeapYear(calendarDate /*, cache */) {
+      // Leap years happen one year before the Julian leap year. Note that this
+      // calendar is based on the Julian calendar which has a leap year every 4
+      // years, unlike the Gregorian calendar which doesn't have leap years on
+      // years divisible by 100 except years divisible by 400.
+      //
+      // Note that we're assuming that leap years in before-epoch times match
+      // how leap years are defined now. This is probably not accurate but I'm
+      // not sure how better to do it.
+      const { year } = calendarDate;
+      return (year + 1) % 4 === 0;
+    },
+    monthsInYear(/* calendarDate */) {
+      return 13;
+    },
+    minimumMonthLength(calendarDate) {
+      const { month } = calendarDate;
+      // Ethiopian/Coptic calendars have 12 30-day months and an extra 5-6 day 13th month.
+      if (month === 13) return this.inLeapYear(calendarDate) ? 6 : 5;
```

Comment:
The 8 day increment logic in the general helper doesn't matter for correctness, only performance, right?

Reply:
Correct. Only perf.